RadarSkillPlot.py

```diff
@@ -0,0 +1,296 @@
+
+    log = logging.getLogger('RadarTest')
+    log.addHandler(file_handler)
+
+    if is_success:     # success
+        print('\033[;32m' + string + '\033[m')
+    elif not is_success:   # error
+        log.error('\n\033[;31m' + string + '\033[m\n')    # outputs as stderr
+    else:
+        print(string)
+
+
+def critical_checks(config_path_, is_demo_=False):
+    passed_ = True
+    config_yaml = None
+    yaml_parsed_ = None
+    skills_by_date_ = None
+
+    # Check that the config file exists, and read content from it.
+    if is_demo_:
+        config_yaml = example_yaml
+    else:
+        if os.path.isfile(config_path_):
+            with open(config_path_, 'r') as f:
+                config_yaml = f.read()
+        else:
+            print_success_error('Invalid config path provided. Chart will not be plotted', is_success=False)
+            passed_ = False
+
+    # Attempt to parse the config file content as YAML content.
+    if passed_:
+        try:
+            yaml_parsed_ = yaml.load(config_yaml)
+        except y_scanner.ScannerError as exception_:
+            print_success_error('Invalid YAML format in config file. Chart will not be plotted \nError message: \n\n' +
+                                str(exception_), is_success=False)
+            passed_ = False
+
+    # Search YAML top level for "Skills By Date", which contains the core information.
+    if passed_:
+        if 'Skills By Date' in yaml_parsed_:
+            skills_by_date_ = yaml_parsed_['Skills By Date']
+        if type(skills_by_date_) is not list:
+            print_success_error('Config "Skills By Date" key missing/invalid. Chart will not be plotted',
+                                is_success=False)
+            passed_ = False
+
+    return passed_, yaml_parsed_, skills_by_date_
+
+
+def extract_data(skills_by_date_):
+    # Extract the data from the YAML structure into an un-pivoted list of dates and skill levels.
+    un_pivoted_ = []
+    color_dict_ = {}
+    for index_, date_dict in enumerate(skills_by_date_):
+        date_value = None  # date_value needs to be reset as it is used for rollback in case of an exception.
+
+        try:
+            date_value = date_dict['Date'].strftime('%Y-%m-%d')
+            skills_list = date_dict['Skills']
+
+            # Supplying a colour for the date is optional and can fall back on default colours.
+            # Colours are stored in a separate dated dictionary as they do not belong in the 2D data frame.
+            color_value = date_dict.get('Colour', 'default')
+            color_dict_[date_value] = color_value
+
+            for k, v in skills_list.items():
+                if type(v) not in [int, float, complex]:
+                    v = None
+                    print('Date: %s  Skill: %s  is non-numeric - point will not be plotted' % (date_value, str(k)))
+                un_pivoted_.append([date_value, k, v])
+
+        except (KeyError, ValueError, AttributeError):
```

Comment:
I don't like that the original exception content is lost here.
Otherwise you risk assuming that you understand "why" a given exception type happened, and you also lose the information about where in the call stack it actually happened.
Much better to do it like line 113-115, where you include the original exception message within your own message.